sklearn/neighbors/_nearest_centroid.py

```diff
@@ -255,7 +314,100 @@ def predict(self, X):
         """
         check_is_fitted(self)
 
-        X = self._validate_data(X, accept_sparse="csr", reset=False)
-        return self.classes_[
-            pairwise_distances_argmin(X, self.centroids_, metric=self.metric)
-        ]
+        if len(np.unique(self.class_priors_, return_counts=True)[0]) == 1:
+            X = self._validate_data(X, accept_sparse="csr", reset=False)
```

Comment:
This `if` block does not call the `_decision_function` so I don't think there's double call to the `_validate_data`, only `else` block call it.

Reply:
This is my point for adding a comment since this is surprising why to call the validation here and not in other case (you need to look at the `_decision_function` to know about it). So a small comment could be useful to not have to navigate to understand the code.

Reply:
Oh my bad, I thought you mean adding a comment here on the PR review, not in the code. Sure.